### ISSUE TITLE
ProxyController base class

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#1737]
Proxy workers currently derive of RouterController

and hence register a lot of management API which is router specific and should not be there.

Probably ProxyController should inherit from NativeWorkerController instead.

```console
2020-03-29T21:55:30+0200 [Controller  30083] Order node to start "Proxy pxy_3" ..
2020-03-29T21:55:30+0200 [Controller  30083] Starting proxy worker pxy_3 <crossbar.node.controller.NodeController.start_worker>

2020-03-29T21:55:32+0200 [Proxy       30104] Starting worker "pxy_3" for node "lojack1" on realm "crossbar" with personality "edge" <crossbar.worker.proxy.ProxyController>
2020-03-29T21:55:32+0200 [Proxy       30104] Running as PID 30104 on CPython-EPollReactor
2020-03-29T21:55:32+0200 [Proxy       30104] Entering event reactor ...
2020-03-29T21:55:32+0200 [Controller  30083] Router.attach(session=6758377946764494)
2020-03-29T21:55:32+0200 [Controller  30083] Router.attach(session=6758377946764494): attached session 6758377946764494 to router realm "crossbar"
2020-03-29T21:55:32+0200 [Controller  30083] WAMP session statistics DISABLED
2020-03-29T21:55:32+0200 [Proxy       30104] Proxy worker "pxy_3" session 6758377946764494 initializing ..
2020-03-29T21:55:32+0200 [Proxy       30104] Registered 52 management procedures on realm "crossbar"
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.add_pythonpath
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_cpu_affinity
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_cpu_count
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_cpus
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_manhole
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_node_id
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_process_info
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_process_monitor
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_process_stats
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_profile
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_profilers
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_pythonpath
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_component
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_components
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm_link
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm_links
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm_role
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm_roles
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realm_stats
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_realms
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_transport
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_router_transports
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.get_web_transport_service
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.kill_by_authid
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.set_cpu_affinity
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.set_node_id
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.set_process_stats_monitoring
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.shutdown
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_manhole
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_profiler
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_proxy_connection
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_proxy_route
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_proxy_transport
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_router_component
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_router_realm
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_router_realm_link
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_router_realm_role
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_router_transport
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.start_web_transport_service
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.started
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_manhole
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_proxy_transport
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_router_component
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_router_realm
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_router_realm_link
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_router_realm_role
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_router_transport
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.stop_web_transport_service
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.trigger_gc
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.uptime
2020-03-29T21:55:32+0200 [Proxy       30104]   crossbar.worker.pxy_3.utcnow
2020-03-29T21:55:32+0200 [Controller  30083] Ok, node has started Proxy pxy_3
```

**Comment by oberstet** [comment]
So my concrete proposal would be a one line change to `class ProxyController(NativeWorkerController)`. @meejah do you think that would work?

**Comment by meejah** [comment]
We still do want the transport setup stuff (`start_router_transport`) .. which probably means some "related" ones as well. These could be factored out (but, more than a one-liner I think).

**Comment by oberstet** [comment]
fixed via #1755
